### DISCUSSION TITLE
x-show modifiers

### DISCUSSION THREAD

**Discussion by webmaxx** [alpinejs/alpine#2831] [3. Feature ideas]
At the moment x-show == toggle "display: none"

But in this case, the height of the element becomes 0, which is not always what you need.

Is it possible to add modifiers or suffixes?

```
x-show:visibility == toggle "visibility: hidden"
x-show:opacity == toggle "opacity: 0"
```

```
x-show.visibility == toggle "visibility: hidden"
x-show.opacity == toggle "opacity: 0"
```

**Comment by SimoTod** [comment]
A bit more verbose but if you don't have nested x-show's or complicate transitions you can do
`:style="show ? 'visibility: visible' : 'visibility: hidden'"`
`:style="show ? 'opacity: 1' : 'opacity: 0'"`

**Reply by webmaxx** [reply]
I agree. That's also possible. But in this case x-transition does not work

**Comment by dannyuk1982** [comment]
+1 to this, this is often a problem, especially when showing/hiding content i.e. in carousels, using `display: none` over `visibility: visible` or `opacity: 0` means the height of the element changes, causing layout shift.

**Comment by stepanjakl** [comment]
+1 for having a x-show modifier such as `x-show.visibility` or similar that also works with x-transition
